pkg/prometheus/operator.go

```diff
@@ -969,18 +968,22 @@ func (c *Operator) sync(key string) error {
 }
 
 func createSSetInputChecksum(p monitoringv1.Prometheus, c Config, ruleConfigMapNames []string) (string, error) {
```

Comment:
minor nit: the function states it creates a checksum, while we calculate a hash. While similar in nature, the purpose of a checksum (error detection) vs. a hash (uniformity) is slightly different. Maybe we can change the func name to reflect this fact?

Reply:
what does the extra `S` mean in this func name?

Reply:
Good idea. Changed across the project.

Reply:
I guess `createSSetInputChecksum` expands to `createS[tateful]SetInputChecksum`. How about introducing:

```
type statefulSetConfig struct {
  prom monitoringv1.Prometheus
  operatorConfig Config
  ruleConfigMapNames []string
}
```
and implement a `Hash() (string, error)` as well as a `StatefulSet() (*appsv1.StatefulSet, error)`  method which would also reduce the amount of parameters: https://github.com/coreos/prometheus-operator/blob/cc98874a6df78cfb0d0fdb8edac3ae8d0ef68739/pkg/prometheus/statefulset.go#L80-L86

Reply:
I went through a couple of iterations on this idea, but ended up bursting the scope of this PR. I will follow up with you on this @s-urbaniak if you don't mind.

> what does the extra S mean in this func name?

@squat `SSet` stands for a StatefulSet.